src/s3/activation.ts

```diff
@@ -27,6 +28,9 @@ export async function activate(ctx: ExtContext): Promise<void> {
         vscode.commands.registerCommand('aws.s3.copyPath', async (node: S3FolderNode | S3FileNode) => {
             await copyPathCommand(node)
         }),
+        vscode.commands.registerCommand('aws.s3.presignedURL', async (node: S3FileNode) => {
+            await presignedURLCommand(node)
```

Comment:
can we go further? Would it be useful to have a command that takes user input when run from the command palette, instead of the AWS Explorer? I.e. it prompts user and takes a URL.

Reply:
Yes this would be preferred. The command should not assume that a node exists (but if it does, use that for the URL)

Reply:
Can you elaborate, I did not understand what you meant, do you mean to open the URL inside VSCode?

Reply:
I mean expose the command as something that can be run via ctrl-shift-p or cmd-shift-p. So user can run `AWS: Generate Presigned S3 URL`, then a prompt appears and accepts a URL.

This is not a new command, it is the same command, but it checks the argument `node`, if `node` is a string it treats it as such, else it expects a Explorer node.

Reply:
Personally, I feel like the right click option is fine for a P0 release, unless you think a lot of people will be jumping at the bit to copy/paste an ARN into VS Code.

Maybe implement that in a second PR, at least?

Reply:
> Maybe implement that in a second PR, at least?

Ok. It should be very little additional code.